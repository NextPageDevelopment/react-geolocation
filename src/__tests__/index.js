/* eslint-env jest */
// eslint-disable-next-line no-unused-vars
import React from 'react'
import Geolocation from '../index'
import renderer from 'react-test-renderer'
const TestUtils = require('react-dom/test-utils')

describe('react-geolocation', () => {
  test('renders', () => {
    const tree = renderer.create(<Geolocation lazy />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders with children', () => {
    const renderCallback = result => {
      return (
        <pre>
          {JSON.stringify(result)}
        </pre>
      )
    }
    const tree = renderer
      .create(
        <Geolocation lazy>
          {renderCallback}
        </Geolocation>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('can render with no children', () => {
    TestUtils.renderIntoDocument(<Geolocation lazy />)
  })
})
