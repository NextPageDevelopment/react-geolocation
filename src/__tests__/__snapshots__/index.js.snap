// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-geolocation renders 1`] = `null`;

exports[`react-geolocation renders with children 1`] = `null`;
